tokio/src/sync/watch.rs

```diff
@@ -380,7 +380,17 @@ mod state {
     impl Version {
         /// Get the initial version when creating the channel.
         pub(super) fn initial() -> Self {
-            Version(0)
+            // We start counting at 2 so that we can mark the
+            // Receiver as unread on creation.
+            Version(2)
```

Comment:
```suggestion
            // The initial version is 1 so that `mark_unseen` can decrement by one.
            // (The value is 2 due to the closed bit.)
            Version(2)
```